### ISSUE TITLE
Huge battery drain / Random crashes

### ISSUE THREAD

**Issue by deathline94** [hiddify/hiddify-next#689]
### Search first

- [X] I searched and no similar issues were found

### What Happened?

Basically two different bugs here :
I had a mere 40% battery drain on the latest version of android 14. With battery optimizer off for the app.
Any logs or logcats or dmesg would be provided upon further request.

The second bug is random crashes. Basically, you connect the app and let the phone sit aside. There's a good chance that when you pick it up after an hour, the app won't open or stays suspended or won't be responsive until you force close it and re-open it.
I tried this with and without battery optimizer and memory limiter.

### Minimal Reproducible Example (MRE)

Basically, just hit connect !

### Version

v1.0.0

### Platform/OS

Android

### Are you willing to submit a PR? If you know how to fix the bug.

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by lymanjre** [comment]
It s repetitive
